pkg/common/helper.go

```diff
@@ -134,25 +134,27 @@ func RemoveANSIColorsFromString(s string) string {
 
 // RetryUntilSuccessful calls callback every interval for duration until it returns true
 func RetryUntilSuccessful(duration time.Duration, interval time.Duration, callback func() bool) error {
-	return retryUntilSuccessful(duration, interval, func() (bool, error) {
+	return retryUntilSuccessful(duration, interval, func(int) (bool, error) {
 
 		// callback results indicate whether to retry
 		return !callback(), nil
 	})
 }
 
 // RetryUntilSuccessfulOnErrorPatterns calls callback every interval for duration as long as error pattern is matched
-func RetryUntilSuccessfulOnErrorPatterns(duration time.Duration,
+func RetryUntilSuccessfulOnErrorPatterns(
+	duration time.Duration,
 	interval time.Duration,
 	errorRegexPatterns []string,
-	callback func() string) error {
+	callback func(retryCounter int) (string, error)) error {
```

Comment:
the `retryCounter` is to be checked inside the callback only if needed, but maybe add a couple of words on it in the docstring above?

Reply:
the retrycounter is just for logging purposes atm, mainly to show how many retries were made given the time the retry until successful run